### PR TITLE
chore(docs): added more docs for core.ts

### PR DESCRIPTION
**Pull request by mhevery**

### CODE REVIEW

**Comment by petebacondarwin** [comment]
#3003 fixes problems highlighted by this PR

**Comment by petebacondarwin** [comment]
The PR will need to be rebased on master since the second commit added here is not the same as the one on master

File: modules/angular2/core.ts [hunk 1]
```diff
@@ -15,17 +15,16 @@ export {DirectiveResolver} from 'angular2/src/core/compiler/directive_resolver';
 export {Compiler} from 'angular2/src/core/compiler/compiler';
 
 export {AppViewManager} from 'angular2/src/core/compiler/view_manager';
-export {QueryList} from 'angular2/src/core/compiler/query_list';
-export {BaseQueryList} from 'angular2/src/core/compiler/base_query_list';
+export {DynamicComponentLoader} from 'angular2/src/core/compiler/dynamic_component_loader';
```
Comment (tbosch) [inline modules/angular2/core.ts]: Maybe group all exports of `dynamic_component_loader` again?

**Comment by tbosch** [comment]
@mhevery could we wait a little bit with adding `HostViewRef`? I have my big PR for the render change almost in (hopefully tomorrow). We should probably also add an interface for `EmbeddedViewRef` and also split the methods in `ViewContainerRef` to deal with one or the other. Maybe in a separate PR?

File: modules/angular2/src/core/compiler/view_manager.ts [hunk 2]
```diff
@@ -138,7 +138,7 @@ export class AppViewManager {
    * ```
    */
   createRootHostView(hostProtoViewRef: ProtoViewRef, overrideSelector: string,
-                     injector: Injector): ViewRef {
+                     injector: Injector): HostViewRef {
```
Comment (tbosch) [inline modules/angular2/src/core/compiler/view_manager.ts]: Please also use `HostViewRef` in `createHostViewInContainer`, and also in `ViewContainerRef.createHostView`

Comment (mhevery) [inline modules/angular2/src/core/compiler/view_manager.ts]: done

**Comment by mhevery** [comment]
c83a3f3372a0a7d55dd95dced89665d24607fbb1

**Pull request closed without merging**